tracer/src/Datadog.Trace/SpanExtensions.Core.cs

```diff
@@ -0,0 +1,39 @@
+// <copyright file="SpanExtensions.Core.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+#if !NETFRAMEWORK
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Datadog.Trace.AppSec;
+using Datadog.Trace.AppSec.Coordinator;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc.Abstractions;
+using Microsoft.AspNetCore.Routing;
+using Microsoft.AspNetCore.Routing.Template;
+
+namespace Datadog.Trace
+{
+    /// <summary>
+    /// Extension methods for the <see cref="ISpan"/> interface
+    /// </summary>
+    public static partial class SpanExtensions
+    {
+        private static void RunBlockingCheck(Span span, string userId)
+        {
+            var security = Security.Instance;
+
+            if (security.Settings.Enabled)
+            {
+                var httpContext = CoreHttpContextStore.Instance.Get();
+
+                security.CheckUser(httpContext, span, userId);
```

Comment:
are we sure httpcontext can never be null here? if we didn't go through the right stage of the Observer yet? but I guess `SetUserDetails` cant be called before `OnHttpRequestStart`?